frontend/src/metabase/public/containers/PublicOrEmbeddedDashboard/PublicOrEmbeddedDashboardView.tsx

```diff
@@ -0,0 +1,242 @@
+  draftParameterValues,
+  setParameterValue,
+  setParameterValueToDefault,
+  dashboardId,
+  bordered,
+  titled,
+  theme,
+  hideParameters,
+  hideDownloadButton,
+  navigateToNewCardFromDashboard,
+  slowCards,
+  cardTitled,
+}: {
+  dashboard: Dashboard | null;
+  hasNightModeToggle?: boolean;
+  isFullscreen: boolean;
+  isNightMode: boolean;
+  onFullscreenChange: DashboardFullscreenControls["onFullscreenChange"];
+  onNightModeChange: EmbedThemeControls["onNightModeChange"];
+  onRefreshPeriodChange: DashboardRefreshPeriodControls["onRefreshPeriodChange"];
+  refreshPeriod: RefreshPeriod;
+  setRefreshElapsedHook: DashboardRefreshPeriodControls["setRefreshElapsedHook"];
+  selectedTabId: SelectedTabId;
+  parameters: UiParameter[];
+  parameterValues: Record<string, ParameterValueOrArray>;
+  draftParameterValues: Record<string, ParameterValueOrArray | null>;
+  setParameterValue: HandleThunkActionCreator<
+    typeof setParameterValueDashboardAction
+  >;
+  setParameterValueToDefault: HandleThunkActionCreator<
+    typeof setParameterValueToDefaultDashboardAction
+  >;
+  dashboardId: DashboardId;
+  bordered: boolean;
+  titled: boolean;
+  theme: DisplayTheme;
+  hideParameters: EmbedHideParameters;
+  hideDownloadButton: EmbedHideDownloadButton;
+  navigateToNewCardFromDashboard?: (
+    opts: NavigateToNewCardFromDashboardOpts,
+  ) => void;
+  slowCards: Record<number, boolean>;
+  cardTitled: boolean;
+}) {
+  const buttons = !isWithinIframe()
+    ? getDashboardActions({
+        dashboard,
+        hasNightModeToggle,
+        isFullscreen,
+        isNightMode,
+        onFullscreenChange,
+        onNightModeChange,
+        onRefreshPeriodChange,
+        refreshPeriod,
+        setRefreshElapsedHook,
+        isPublic: true,
+      })
+    : [];
+
+  const visibleDashcards = (dashboard?.dashcards ?? []).filter(
+    dashcard => !isActionDashCard(dashcard),
+  );
+
+  const dashboardHasCards = dashboard && visibleDashcards.length > 0;
+
+  const tabHasCards =
+    visibleDashcards.filter(
+      (dc: DashboardCard) => dc.dashboard_tab_id === selectedTabId,
+    ).length > 0;
+
+  const hiddenParameterSlugs = getHiddenParameterSlugs({
+    parameters,
+    dashboard,
+    selectedTabId,
+  });
+
+  return (
+    <EmbedFrame
+      name={dashboard && dashboard.name}
+      description={dashboard && dashboard.description}
+      dashboard={dashboard}
+      parameters={parameters}
+      parameterValues={parameterValues}
+      draftParameterValues={draftParameterValues}
+      hiddenParameterSlugs={hiddenParameterSlugs}
+      setParameterValue={setParameterValue}
+      setParameterValueToDefault={setParameterValueToDefault}
+      enableParameterRequiredBehavior
+      actionButtons={
+        buttons.length > 0 ? <div className={CS.flex}>{buttons}</div> : null
+      }
+      dashboardTabs={
+        dashboard?.tabs &&
+        dashboard.tabs.length > 1 && <DashboardTabs dashboardId={dashboardId} />
+      }
```

Comment:
nit: actionButtons uses `... ? element : null` while dashboardTabs uses `... && element`. could we do the same for both